crates/cli/src/introspection/validation_schema.rs

```diff
@@ -20,42 +19,34 @@ pub async fn get_metadata_from_validation_schema(
     config: &MongoConfig,
 ) -> Result<BTreeMap<String, Schema>, MongoAgentError> {
     let db = config.client.database(&config.database);
-    let collections_cursor = db.list_collections(None, None).await?;
-
-    let schemas: Vec<WithName<Schema>> = collections_cursor
-        .into_stream()
-        .map(
-            |collection_spec| -> Result<WithName<Schema>, MongoAgentError> {
-                let collection_spec_value = collection_spec?;
-                let name = &collection_spec_value.name;
-                let schema_bson_option = collection_spec_value
-                    .options
-                    .validator
-                    .as_ref()
-                    .and_then(|x| x.get("$jsonSchema"));
-
-                match schema_bson_option {
-                    Some(schema_bson) => {
-                        from_bson::<ValidatorSchema>(schema_bson.clone()).map_err(|err| {
-                            MongoAgentError::BadCollectionSchema(
-                                name.to_owned(),
-                                schema_bson.clone(),
-                                err,
-                            )
-                        })
-                    }
-                    None => Ok(ValidatorSchema {
-                        bson_type: BsonType::Object,
-                        description: None,
-                        required: Vec::new(),
-                        properties: IndexMap::new(),
-                    }),
-                }
-                .map(|validator_schema| make_collection_schema(name, &validator_schema))
-            },
-        )
-        .try_collect::<Vec<WithName<Schema>>>()
-        .await?;
+    let mut collections_cursor = db.list_collections(None, None).await?;
+
+    let mut schemas: Vec<WithName<Schema>> = vec![];
+
+    while let Some(collection_spec) = collections_cursor.try_next().await? {
+        let name = &collection_spec.name;
+        let schema_bson_option = collection_spec
+            .options
+            .validator
+            .as_ref()
+            .and_then(|x| x.get("$jsonSchema"));
+
+        match schema_bson_option {
+            Some(schema_bson) => {
+                let validator_schema =
+                    from_bson::<ValidatorSchema>(schema_bson.clone()).map_err(|err| {
+                        MongoAgentError::BadCollectionSchema(
```

Comment:
If the collection has a something in `$jsonSchema` that can't be parsed as a `ValidatorSchema` then we treat this as an error. Perhaps we should just log this and continue, in which case we would fall back to sampling for this collection. But I think it's probably better to treat it as an error.